xarray/core/combine.py

```diff
@@ -421,6 +45,71 @@ def _infer_tile_ids_from_nested_list(entry, current_pos):
         yield current_pos, entry
 
 
+def _infer_concat_order_from_coords(datasets):
+
+    concat_dims = []
+    tile_ids = [() for ds in datasets]
+
+    # All datasets have same variables because they've been grouped as such
+    ds0 = datasets[0]
```

Comment:
Do we always know that `datasets` is non-empty?

Let's make sure that `combine_auto()` and `combine_manual()` do something reasonable in the case of empty input, either by returning an empty Dataset (preferred) or raising an error.

Reply:
I had to special case this in `_manual_combine`, but I've added tests to ensure that an empty Dataset is always returned.